srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -235,6 +254,34 @@ private void retrieveLabs(final int dfn, final Patient patient) throws ParseExce
         }
     }
     
+    private void retrieveHealthFactors(final int dfn, final Patient patient)
+    {
+        try
+        {
+            patient.getHealthFactors().clear();
+            final List<String> rpcResults = fProcedureCaller.doRpc(
+                    fDuz, RemoteProcedure.GET_HEALTH_FACTORS, String.valueOf(dfn));
+            // Now that we have all of the health factors, filter out any that are not present
+            // in the list provided by the NSO.
+            final Iterator<String> iter = rpcResults.iterator();
+            while(iter.hasNext())
+            {
+                final String healthFactor = iter.next();
+                final String[] factorSplitArray = healthFactor.split("\\^");
+                if(VALID_HEALTH_FACTORS.get(factorSplitArray[1]) != null)
```

Comment:
This algorithm is certainly improved, but it could be simpler. You only ever use VALID_HEALTH_FACTORS to check if it contains items, right? In that case a Map is needlessly complex. What simpler data structure could you use?